bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/BoschHttpClient.java

```diff
@@ -0,0 +1,139 @@
+package org.openhab.binding.boschshc.internal;
+
+import static org.eclipse.jetty.http.HttpMethod.GET;
+
+import java.nio.charset.StandardCharsets;
+import java.security.KeyStoreException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateEncodingException;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+public class BoschHttpClient extends HttpClient {
+
+    private static final int MAX_PAIR_TRIES = 12;
+    
+	private final Logger logger = LoggerFactory.getLogger(BoschHttpClient.class);
+
+	private String ipAddress;
+	private String systempassword;
+
+	public BoschHttpClient(String ipAddress, String systempassword, SslContextFactory sslContextFactory) {
+		super(sslContextFactory);
+		this.ipAddress = ipAddress;
+		this.systempassword = systempassword;
+	}
+
+	private String getCertFromSslContextFactory() throws KeyStoreException, CertificateEncodingException {
+		Certificate cert = this.getSslContextFactory().getKeyStore().getCertificate(BoschSslUtil.OSS_OPENHAB_BINDING);
+		return new String(Base64.getEncoder().encode(cert.getEncoded()));
+	}
+
+	public void checkAccessAndPairIfNecessary() throws InterruptedException {
+        // test if pairing is needed
+   		int counter = 0;
+       	boolean accessPossible = isAccessPossible();
+       	while (!accessPossible) {
+			// sleep some seconds after every try, except the first one
+   			if (counter > 0) {
+   				logger.info("Last Pairing failed, starting retry number {}/{} in few seconds", counter, MAX_PAIR_TRIES);
+       			Thread.sleep(15000);
+   			} else {
+   				logger.info("Pairing needed, because access to Bosch SmartHomeController not possible.");
```

Comment:
Wondering how we can do this in the long run when we want to exclusively rely on the UI rather than console + configuration files. I have absolutely no experience on how this works. Perhaps one of the OH developers can help with that or we look how it is done in the Hue binding. There, you also need to press a button to init the connection.